src/CodeFormatter.cs

```diff
@@ -30,6 +30,8 @@ internal static class CodeFormatter
             new CharsetFormatter(),
         }.ToImmutableArray();
 
+        private static readonly Dictionary<ProjectId, AnalyzerConfigSet> s_analyzerConfigSets = new Dictionary<ProjectId, AnalyzerConfigSet>();
```

Comment:
💡 Consider using `ConditionalWeakTable<Project, AnalyzerConfigSet>` to avoid unbounded memory growth if used in a long-lived process.

Reply:
skeptical that this will matter for a commandline tool

---
In reply to: [393801704](https://github.com/dotnet/format/pull/590#discussion_r393801704) [](ancestors = 393801704)